platform/src/main/java/org/hillview/storage/CassandraSSTableLoader.java

```diff
@@ -257,51 +367,89 @@ public ITable load() {
         }
     }
 
     /** Lazy and non-lazy loader will call this function to load specific column marked by columnToLoad */
-    private List<IAppendableColumn> loadColumns(boolean[] columnToLoad, Stream<UnfilteredRowIterator> partitions, int size) {
-        List<IAppendableColumn> columns = createColumns(columnToLoad, size);
-        // Iterating each item inside the table
-        partitions.forEach(partition -> {
-            while (partition.hasNext()) {
-                Unfiltered unfiltered = partition.next();
-                if (unfiltered instanceof Row) {
-                    Row row = (Row) unfiltered;
-                    int i = 0;
-                    int currentColumn = 0;
-                    Object value;
-                    IAppendableColumn col;
-                    // Extracting the value for each column from a single row
-                    for (ColumnData cd : row) {
-                        // Filter the column to only load the one marked true by columnToLoad[]
-                        if (columnToLoad[i]) {
-                            if (cd.column().isSimple()) {
-                                Cell cell = (Cell) cd;
-                                AbstractType<?> cellType = cell.column().cellValueType();
-                                value = cellType.getSerializer().deserialize(cell.value());
-                            } else {
-                                // Hillview won't process the complex data
-                                throw new RuntimeException("Hillview can't convert complex data");
+    private static void loadColumns(UnfilteredRowIterator partition,
+        List<IAppendableColumn> columns, boolean[] columnToLoad, List<CassandraType> columTypes){
+        Unfiltered unfiltered = partition.next();
+        if (unfiltered instanceof Row) {
+            Row row = (Row) unfiltered;
+            int i = 0;
+            int currentColumn = 0;
+            Object value;
+            IAppendableColumn col;
+            // Extracting the value for each column from a single row
+            for (ColumnData cd : row) {
+                // Filter the column to only load the one marked true by columnToLoad[]
+                if (columnToLoad[i]) {
+                    if (cd.column().isSimple()) {
+                        col = columns.get(currentColumn);
+                        AbstractType<?> cellType = cd.column().cellValueType();
+                        value = cellType.getSerializer().deserialize(((Cell) cd).value());
+                        if(value == null) {
+                            col.appendMissing();
+                        } else {
+                            switch (columTypes.get(i)) {
+                                case STRING:
+                                    col.append(value.toString());
+                                    break;
+                                case CQLSTRING:
+                                    col.append(cellType.getSerializer()
```

Comment:
how is this different from just value.toString()?

Reply:
When it comes to converting time and date, the result is not as expected. For example:
- Original Cassandra format
        date : 2020-07-14
        time : 13:30:23.123000000
- Use normal serializer 
        date : -2147465191    
        time : 48623123000000
- Use serializer.toCQLLiteral
        date : 2020-07-14
        time : 13:30:23.123000000

That's why I create a new type as CQLSTRING